File: modules/api/src/main/java/net/countercraft/movecraft/events/CraftPilotEvent.java
package net.countercraft.movecraft.events;

import net.countercraft.movecraft.craft.Craft;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

/**
 * Called whenever a craft is piloted
 * @see Craft
 */
@SuppressWarnings("unused")
public class CraftPilotEvent extends CraftEvent{
    @NotNull private final Reason reason;
    private static final HandlerList HANDLERS = new HandlerList();

    public CraftPilotEvent(@NotNull Craft craft, @NotNull Reason reason){
        super(craft);
        this.reason = reason;
    }

    @NotNull
    public Reason getReason() {
        return reason;
    }

    public enum Reason{
        SUB_CRAFT,PLAYER,FORCE
    }

    @Override
    public HandlerList getHandlers() {
        return HANDLERS;
    }

    @SuppressWarnings("unused")
    public static HandlerList getHandlerList() {
        return HANDLERS;
    }
}

File: src/main/kotlin/dev/cccm5/movecraft/event/CraftPilotEvent.kt
package dev.cccm5.movecraft.event

import dev.cccm5.movecraft.craft.Craft
import dev.cccm5.movecraft.CraftStateManager
import org.bukkit.event.Cancellable
import org.bukkit.event.HandlerList

/**
 * Called whenever a [Craft] is piloted
 * The term piloted is used to indicate that a craft is being added to the [CraftStateManager]
 */
class CraftPilotEvent(craft: Craft): CraftEvent(craft), Cancellable {
    private var cancelled = false
    override fun setCancelled(cancel: Boolean) {
        this.cancelled = cancel
    }

    override fun isCancelled(): Boolean {
        return this.cancelled
    }

    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        @JvmStatic private val handlerList = HandlerList()
        @JvmStatic fun getHandlerList() = handlerList
    }
}